domain/src/main/java/org/oppia/domain/audio/AudioPlayerController.kt

```diff
@@ -0,0 +1,74 @@
+package org.oppia.domain.audio
+
+import android.content.Context
+import android.media.MediaPlayer
+import android.net.Uri
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class AudioPlayerController @Inject constructor(val context: Context) {
+
+  class PlayStatus(val type: String, val value: Int)
```

Comment:
Discussed in meeting: since LiveData observers can miss updates (e.g. during configuration changes), we need to model this based on absolute state. I think that means we need to include position & duration each time.